### PR TITLE
post likes

### PR DESCRIPTION
**Pull request by pereslop**

### CODE REVIEW

File: app/controllers/account/users/confirmations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+class Account::Users::ConfirmationsController < Devise::ConfirmationsController
```
Comment (Snick555) [inline app/controllers/account/users/confirmations_controller.rb]: why do you need this controller?

File: app/controllers/account/users/omniauth_callbacks_controller.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+class Account::Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
```
Comment (Snick555) [inline app/controllers/account/users/omniauth_callbacks_controller.rb]: why do you need this controller?

File: app/controllers/account/users/passwords_controller.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
+class Account::Users::PasswordsController < Devise::PasswordsController
```
Comment (Snick555) [inline app/controllers/account/users/passwords_controller.rb]: why do you need this controller?

File: app/controllers/account/users/registrations_controller.rb [hunk 4]
```diff
@@ -0,0 +1,60 @@
+class Account::Users::RegistrationsController < Devise::RegistrationsController
```
Comment (Snick555) [inline app/controllers/account/users/registrations_controller.rb]: why do you need this controller?

File: app/controllers/account/users/unlocks_controller.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
+class Account::Users::UnlocksController < Devise::UnlocksController
```
Comment (Snick555) [inline app/controllers/account/users/unlocks_controller.rb]: why do you need this controller?

File: app/controllers/account/users/sessions_controller.rb [hunk 6]
```diff
@@ -0,0 +1,30 @@
+class Account::Users::SessionsController < Devise::SessionsController
```
Comment (Snick555) [inline app/controllers/account/users/sessions_controller.rb:1]: Please remove all commented code

File: app/controllers/home_controller.rb [hunk 7]
```diff
@@ -1,2 +1,15 @@
 class HomeController < ApplicationController
+  def index
+  end
+
+  def show
+  end
+
+  private
+
+  def collection
```
Comment (Snick555) [inline app/controllers/home_controller.rb]: why do you need these empty methods?

File: app/views/account/users/show.html.slim [hunk 8]
```diff
@@ -0,0 +1,22 @@
+
+.row
+  .col-md-2.text-center
+    h1=  @user.username
+    - if @user.username == current_user.username
```
Comment (Snick555) [inline app/views/account/users/show.html.slim]: Is username unique? What will be if another user with the same username come here?

**Review by Snick555 — CHANGES_REQUESTED** [review]

**Comment by Snick555** [comment]
Please don't push local images.

File: app/controllers/home_controller.rb [hunk 9]
```diff
@@ -1,2 +1,7 @@
 class HomeController < ApplicationController
+  def index
+    if user_signed_in?
```
Comment (Snick555) [inline app/controllers/home_controller.rb]: please add `before_action :authenticate_user!`

File: app/models/user.rb [hunk 10]
```diff
@@ -16,16 +16,21 @@
...
-#  role_id                :integer
+#  role                   :integer          default("user")
 #
 
 class User < ApplicationRecord
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable and :omniauthable
+  has_many :posts, dependent: :destroy
+
+   validates :username, :presence => true,
```
Comment (Snick555) [inline app/models/user.rb]: please use newest syntax `presence: true`

File: db/migrate/20170929085339_create_likes.rb [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+  def change
+    create_table :likes do |t|
+      t.string  :liker_type
+      t.integer :liker_id
+      t.string  :likeable_type
+      t.integer :likeable_id
+      t.datetime :created_at
+    end
+
+    add_index :likes, ["liker_id", "liker_type"],       :name => "fk_likes"
```
Comment (Snick555) [inline db/migrate/20170929085339_create_likes.rb]: why do you use rocket syntax?

File: app/views/account/users/show.html.slim [hunk 12]
```diff
@@ -0,0 +1,19 @@
+.container
+  .row
+    .col-md-3.text-center
+      h1=  @user.username
+      - if @user.email == current_user.email
+      = link_to "Edit profile", edit_user_registration_path, :method => :get
+      - if @user.email == current_user.email
```
Comment (Snick555) [inline app/views/account/users/show.html.slim]: please compare ids

**Review by Snick555 — CHANGES_REQUESTED** [review]

File: db/migrate/20170929085339_create_likes.rb (hunk 11 above)
Comment (pereslop) [reply db/migrate/20170929085339_create_likes.rb]: I just copied it from github gem README)

**Review by Snick555 — APPROVED** [review]

**Pull request merged**